prometheus_client/gc_collector.py

```diff
@@ -4,78 +4,44 @@
 
 import gc
 import os
-import time
 
-from .metrics import Histogram
+from .metrics_core import GaugeMetricFamily
 from .registry import REGISTRY
 
 
 class GCCollector(object):
     """Collector for Garbage collection statistics."""
 
-    def __init__(self, registry=REGISTRY, gc=gc):
-        # To work around the deadlock issue described in
-        # https://github.com/prometheus/client_python/issues/322,
-        # the GC collector is always disabled in multiprocess mode.
-        if 'prometheus_multiproc_dir' in os.environ:
+    def __init__(self, registry=REGISTRY):
+        if not hasattr(gc, 'get_stats'):
             return
+        registry.register(self)
 
-        if not hasattr(gc, 'callbacks'):
-            return
-
-        collected = Histogram(
+    def collect(self):
+        collected = GaugeMetricFamily(
             'python_gc_collected_objects',
```

Comment:
Aren't these all counters?

object_collected and objects_uncollectable would also be more usual.

Reply:
Please explain to me:
''object_collected'' Are you taking about variables names or about metrics names?
Is your question about choosing the type of metrics?

Reply:
The metric names.

Also, the types don't seem right.

Reply:
I renamed the metrics names and redid it with counter type